### PR TITLE
Fixing the Bump Map generating Normal Map path

### PR DESCRIPTION
**Pull request by samcake**
- THis PR fixes the the generation of the normal map from a bump map. THis code is still not perfect but at least should solve the current issues of some objects with bad normal maps.

- I believe that there is an actual bad usage of the Bump map field in the reported probllematic models. THe fbx declare that the  "normal" texture is a BUmp map, i think these are actually true normal maps and should be assigned as such correctly in Maya or BLender.
Turns out that the old path for BUmp map was EXACTLY the same as s normal map, not doing the generation phase. I fixed that.
By chance when a genuine normal map is bound and recognized as a bump map, the generation path applied to it preserve the normal map look.

- Removed an extra call to glTexSubImage2D in the GLtexture::transferMip !

- Fixed a bad texel Format in the GLBackendTexelFormat.

### CODE REVIEW

**Comment by samcake** [comment]
To test:
Compare the normal map look on this crate object using debugFramebuffer.js, it is broken in master, not in this PR:
hifi://playa/1.41276,-8.17774,-32.0502/0,-0.656165,0,0.754618 to see the issue

**Comment by hifi-gustavo** [comment]
Build console output for commit: 637da18e5c3dec4123d62dd1ccf7cbc330a95ad5 available at: https://gist.github.com/a61f966f57f74b511f74cb73b437800a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 920ab88ef373c836345f24dfdea1b40aae60c70a available at: https://gist.github.com/a61f966f57f74b511f74cb73b437800a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.dmg

**Comment by Nex-Pro** [comment]
Tested this and observed a marginal increase in the apparent sharpness of the bump effect.
This PR had brighter specular highlights on the Gnomes and the crate. Also a little sparkle on the wood in a couple of places. See comparison images.
pr7675_normals
4707_normals

**Comment by ZappoMan** [comment]
@Nex-Pro - actually Sam meant the shipping containers...

**Comment by Nex-Pro** [comment]
schippingcontainerbump1
schippingcontainerbump2
My eagle eye and inability to read minds revealed the changes in my first  comment , so bonus @ZappoMan

File: libraries/gpu/src/gpu/GLBackendTexture.cpp [hunk 1]
```diff
@@ -214,8 +214,8 @@ bool GLBackend::GLTexture::isReady() const {
 
 // Move content bits from the CPU to the GPU for a given mip / face
 void GLBackend::GLTexture::transferMip(GLenum target, const Texture::PixelsPointer& mip) const {
-    GLTexelFormat texelFormat = GLTexelFormat::evalGLTexelFormat(_gpuTexture.getTexelFormat(), mip->getFormat());
-    glTexSubImage2D(target, 0, 0, 0, _gpuTexture.getWidth(), _gpuTexture.getHeight(), texelFormat.format, texelFormat.type, mip->readData());		      glTexSubImage2D(target, 0, 0, 0, _gpuTexture.getWidth(), _gpuTexture.getHeight(), texelFormat.format, texelFormat.type, mip->readData());
+    GLTexelFormat texelFormat = GLTexelFormat::evalGLTexelFormat(_gpuTexture.getTexelFormat(), mip->getFormat());
```
Comment (ZappoMan) [inline libraries/gpu/src/gpu/GLBackendTexture.cpp]: remove white space at end of line

**Comment by hifi-gustavo** [comment]
Build console output for commit: fc29397d91d3839cc7c37705f5b5c11203af771a available at: https://gist.github.com/a61f966f57f74b511f74cb73b437800a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7675.dmg

**Comment by ZappoMan** [comment]
This is QA Approved and Dev Approved... but we don't want to merge it until AFTER RC5... because we want alpha users to have time to fix their bad content.

**Pull request merged**